docs/iris/src/developers_guide/release.rst

```diff
@@ -0,0 +1,66 @@
+Release
+*******
+
+A release of Iris is a tag on github in the SciTools organisation master.
+
+
+Release Branch
```

Comment:
I think there might be scope here for a section that explains how we decide what goes in a branch,
In summary (according to my experience):

"Finalising the release content"
  * identify any outstanding PRs / issues we'd like in the new release, and label them with the release milestone
  * announce the intention to make a new release on iris-dev group, listing any outstanding issues (ones tagged with the milestone) and inviting contributions via GitHub
  * cut the branch only when the 'outstanding' list has been discussed + resolved 
    * (or "mostly resolved" : we can change our mind about what needs to go in, if priorities or effort estimates change)